index.js

```diff
@@ -2,22 +2,21 @@ let util = require("util");
 let fs = require("fs");
 let readFile = util.promisify(fs.readFile);
 
-exports.middleware = (manifestPath, { caching } = {}) => {
-	if(caching === undefined) {
-		caching = (process.env.NODE_ENV === "production");
-	}
+let PRODUCTION = process.env.NODE_ENV === "production";
 
+exports.middleware = (manifestPath, { caching = PRODUCTION } = {}) => {
 	// we load the manifest here so we get an error early if the
 	// manifest is not readable/parseable
-	// we do it synchronous, because we don't want to await the middleware
-	let manifest = loadJSONSync(manifestPath);
+	// we do it synchronously because we don't want to await the middleware
+	let manifest = require(manifestPath);
+
+	let assetURL = identifier => lookup(manifest, identifier);
 
 	return async (req, res, next) => {
-		// if caching is disabled, refresh the manifest once per request
-		if(!caching) {
+		if(!caching) { // refresh the manifest once per request
 			manifest = await loadJSON(manifestPath);
```

Comment:
Might as well use `require` here too, given that this should only be used in development mode?